src/actix/server_request.rs

```diff
@@ -112,6 +113,19 @@ pub async fn request_to_event(
         .map_err(actix_web::error::ErrorBadRequest)
 }
 
+/// So that an actix-web handler may take an Event parameter
+impl actix_web::FromRequest for Event {
+    type Config = ();
+    type Error = actix_web::Error;
+    type Future = LocalBoxFuture<'static, std::result::Result<Self, Self::Error>>;
+
+    fn from_request(r: &HttpRequest, p: &mut actix_web::dev::Payload) -> Self::Future {
+        let payload = web::Payload(p.take());
+        let request = r.to_owned();
```

Comment:
Why do you need to own the request? the borrow should be enough to invoke `request_to_event`

Reply:
It was how I satisfied the `'static` lifetime constraint of the `LocalBoxFuture`. Happy to hear of a better way!

```
   Compiling cloudevents-sdk v0.3.1 (/home/jim/src/rust/sdk-rust)
error[E0759]: `req` has an anonymous lifetime `'_` but it needs to satisfy a `'static` lifetime requirement
   --> src/actix/server_request.rs:124:20
    |
122 |     fn from_request(req: &HttpRequest, p: &mut actix_web::dev::Payload) -> Self::Future {
    |                          ------------ this data with an anonymous lifetime `'_`...
123 |         let payload = web::Payload(p.take());
124 |         async move { Ok(request_to_event(req, payload).await?) }.boxed_local()
    |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ ...is captured here, requiring it to live as long as `'static`
```

Reply:
Uhm why in first instance this returns a future with a static lifetime?! Lemme read the details :book:

Reply:
Ok let's go this way now. My feeling is that something smells in my use of lifetimes in `HttpRequestDeserializer`